phoenix-core/src/main/java/org/apache/phoenix/coprocessor/UngroupedAggregateRegionObserver.java

```diff
@@ -1430,4 +1430,17 @@ public InternalScanner run() throws Exception {
         }
         return s;
     }
+
+    /**
+     * roll back after split failed, will isRegionClosingOrSplitting set false,
+     * and then write region will is available
+     * @param ctx
+     * @throws IOException
+     */
+    @Override
+    public void preRollBackSplit(ObserverContext<RegionCoprocessorEnvironment> ctx) throws IOException {
```

Comment:
Do you want `preRollBackSplit` and not `postRollBackSplit`? If we're only about to rollback a split, I would say that, semantically, the region is still splitting until that operation is complete. postRollBackSplit seems more correct to me.

Also want to note: this method is only relevant for HBase 1.x.

Reply:
Thanks @joshelser , you are right, postRollBackSplit seems more correct to me . I will close this MR, and reopen a new MR, which use postRollBackSplit method.
HBase 2.x does not have this issue, because 2.x split failed, by SplitTableRegionProcedure rollbackState method , assign open region, so isRegionClosingOrSplitting is false.

Reply:
@joshelser ,hello, I alter this code that postRollBackSplit. could you please review this?

Reply:
> HBase 2.x does not have this issue, because 2.x split failed, by SplitTableRegionProcedure rollbackState method , assign open region, so isRegionClosingOrSplitting is false.

Well, I think it's more accurate to say "HBase 2.x doesn't apply because splits are executed by the master". This coprocessor would never be invoked, running inside a RegionServer.